tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -898,6 +926,164 @@ def fit_extra(self, hyperparameters, time_limit=None, base_model_names=None, **k
         self.save()
         return self
 
+    def _get_all_fit_extra_args(self):
+        ret = list(self._fit_extra_kwargs_dict().keys()) + list(inspect.signature(self.fit_extra).parameters.keys())
+        ret.remove('kwargs')
+
+        return ret
+
+
+    def _run_pseudolabeling(self, unlabeled_data: pd.DataFrame, max_iter: int,
+                            return_pred_prob: bool = False, **kwargs):
+        """
+        Runs pseudolabeling algorithm using the same hyperparameters and model and fit settings
+        used in original model unless specified by the user. This is an internal function that iteratively
+        self labels unlabeled test data then incorporates all self labeled data above a threshold into training.
+        Will keep incorporating self labeled data into training until validation score does not improve
+
+        Parameters:
+        -----------
+        hyperparameters: dict
+            Dictionary of hyperparameters the model should use
+        unlabeled_data: Extra unlabeled data to assign pseudolabels to and incorporate as
+            extra training data.
+        max_iter: int, default = 5
+            Maximum allowed number of iterations, where in each iteration, the data are pseudolabeled
```

Comment:
the unlabeled data are pseudolabeled

Reply:
it's not labeled yet. It gets labeled in this function